draft-ietf-quic-multipath.md

```diff
@@ -885,6 +885,37 @@ two or more active paths during the connection lifetime. Different applications
 Once the implementation has decided which paths to keep alive, it can do so by sending Ping frames
 on each of these paths before the idle timeout expires.
 
+## Migration
+
+Even when the multipath extension is used, migration can happen.
+E.g. the client's source port and/or IP address may change due to changes
+in network topology or address mappings, such as caused by NAT rebinding.
+In this case the server performs path validation (see {{Section 9 of QUIC-TRANSPORT}}
+and both peers are expected to use new connection IDs during this process.
+
+Even is the path validation is causes by a migration event, when
+the multipath extension is used, the
+endpoint treats the receipt of a PATH_CHALLENGE frame as a new path
+and also initiates path validation to open the path.
+This is important to ensure
+that both peers have the same view on the number of active paths. However,
```

Comment:
If both the 4-tuple and the CID changes, the client should send a path challenge in that packet to explicitly open a new path. Otherwise the server should probably ignore that packet as it can't associate it to an exiting path. Maybe something we should also spell out more clearly.

Reply:
Actually ignoring might not be right to address this change where you restart after an idle period with a new CID and then also the NAT rebinding happens... however, it would be the same procedure as in the case when the CID not changes after a NAT rebinding: the server sends a path challenge frame and that would be the signal for the client to assign a new path and use a new CID + close the old path on the presumedly same 4-tuple from the client's point of view.

Reply:
The node that renewed the CID will deduce the path change from invoming traffic on same (incoming) CID and new 4-tuple. It can at any point abandon the old path or retire the old CID.

Reply:
The problem is that after removing the type we used the sender’s cid in the abandon frame and then this doesn’t work as the client only sees one path. However I think we need to fix that an then recommend to send an abandon frame for the old received cid.